Add 3 papers into v2

fm94 opened the pull request:
AD_histogramBased.json > 2009
ITC_Demystified.json > 2010
ITC_flowStatistics.json > 2016

fm94 committed:
Add files via upload

fm94 committed:
Add files via upload

fm94 committed:
Add files via upload

dcferreira commented on v2_papers/2009/AD_histogramBased.json:
```diff
@@ -0,0 +1,179 @@
+      "type": "",
+      "volume": "",
+      "issue": "",
+      "pages": ""
```
These bibtex fields must not be empty strings. If there's no content for them, remove them.
"type" field is mandatory though.
For the bibtex fields I recommend copying the values from the bibtex provided by google scholar (google for the paper in scholar, and look at the citation button).

dcferreira commented on v2_papers/2010/ITC_Demystified.json:
```diff
@@ -0,0 +1,289 @@
+      "type": "",
+      "volume": "",
+      "issue": "",
+      "pages": ""
```
(same thing as above)
These bibtex fields must not be empty strings. If there's no content for them, remove them.
"type" field is mandatory though.
For the bibtex fields I recommend copying the values from the bibtex provided by google scholar (google for the paper in scholar, and look at the citation button).

dcferreira commented on v2_papers/2010/ITC_Demystified.json:
```diff
@@ -0,0 +1,289 @@
+              "flowDurationMilliseconds"
+            ]
+          },
+          {},
```
Empty feature object is not a valid feature (just remove it).

dcferreira commented on v2_papers/2010/ITC_Demystified.json:
```diff
@@ -0,0 +1,289 @@
+          "tcpAckTotalCount",
+          "tcpUrgTotalCount",
+          "tcpCweTotalCount",
+          "tcpEceTotalCount"
```
``"tcpCweTotalCount"`` (is Cwe a typo for Cwr?) and ``"tcpEceTotalCount"`` don't exist in the IANA spec. I would agree that these should be useful features though, and so they should be written as ``"_tcpEceTotalCount"`` and added to the list of valid features of the verifier (I can take care of that if necessary).

dcferreira commented on v2_papers/2016/ITC_flowStatistics.json:
```diff
@@ -0,0 +1,307 @@
+      "type": "",
+      "volume": "",
+      "issue": "",
+      "pages": ""
```
(same as previous papers)
These bibtex fields must not be empty strings. If there's no content for them, remove them.
"type" field is mandatory though.
For the bibtex fields I recommend copying the values from the bibtex provided by google scholar (google for the paper in scholar, and look at the citation button).

dcferreira commented on v2_papers/2016/ITC_flowStatistics.json:
```diff
@@ -0,0 +1,307 @@
+        "bidirectional": true,
+        "features": [
+          "flowDurationMilliseconds",
+          "first10senders",
```
This is not a IANA feature, so it should start with double underscore (``__``), unless you think the feature is useful for other papers, in which case it should start with a single underscore (``_``) and added to our list of valid features.
Check details on these rules here: https://nta-meta-analysis.readthedocs.io/en/latest/features.html

dcferreira commented on v2_papers/2016/ITC_flowStatistics.json:
```diff
@@ -0,0 +1,307 @@
+              }
+            ]
+          },
+          "first10differences",
```
Same problem as ``first10senders``.

dcferreira reviewed (changes requested):
The proposed changes should make the files pass the verifier.
Additionally, please follow the file naming scheme (``lastname_startoftitle.json``) detailed in https://nta-meta-analysis.readthedocs.io/en/latest/global.html.

dcferreira commented:
Thanks for the contribution, I left some comments.
Sorry if they sound harsh, there definitely are some things in the documentation that are not as clear as in my head, so keeping this record is good so that we know how to improve the documentation :)

fm94 committed:
fixed titel and bibtex

fm94 committed:
Fixed authors format

fm94 committed:
updated

1- fixed title
2- fixed bibtex
3- delete empty feature field
4- fixed typo in Cwr flag (feature)
5- replace "tcpEceTotalCount" by "_tcpEceTotalCount" (need to update the verifier)

fm94 committed:
updated

1- fixed title
2- fixed authors
3- fixed bibtex
4- fixed features ("__first10senders" and "__first10differences")

fm94 committed:
update

removed "pages" field from bibtex

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,285 @@
+      "Kim, Hyun-chul",
+      "Jeong, Jiwoong",
+      "Kim, Chong-kwon",
+      "Kwon, Ted “Taekyoung\"",
```
Not completely sure, but the error on the verifier might be because of these weird quotation marks.
I realize the paper has the quotation marks in the name, but it might be best to just remove them.

dcferreira replied:
I just checked myself, the quotation marks are not the problem the verifier has. However, I still think they should just be removed.

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,285 @@
+    "pages_number_of": 12,
+    "bibtex": {
+      "type": "inproceedings",
+    },
```
The problem with the verifier is here. You can't have a comma after the last item in an object (meaning, this isn't valid JSON).
Even then, I'd add these fields:
"booktitle": "Proceedings of the 6th International COnference",
"organization": "ACM"

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,285 @@
+          "tcpPshTotalCount",
+          "tcpAckTotalCount",
+          "tcpUrgTotalCount",
+          "tcpCwrTotalCount",
```
``tcpCwrTotalCount`` is still not a IANA feature. Need an underscore at the start of the name.

dcferreira reviewed (changes requested):
These changes should pass the verifier.

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,285 @@
+      "type": "inproceedings",
+    },
+    "access_open": true,
+    "curated_by": "Meghdouri. f.",
```
I guess this is a typo, the other papers have "Meghdouri, f." (comma instead of period)

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+    "feature_selections": [
+      {
+        "name": "constant-removal",
+        "type": "feature_reduction",
```
This feature selection method seems to me like a ``filter``, and not a ``feature_reduction``.

fm94 replied:
It is kind of a filter but they used it to reduce the dimensionality --number of features--

dcferreira replied:
Filter feature selection always reduces dimensionality (unless the filter is the trivial identity).
``feature_reduction`` refers to methods that transform the space in such a way that the feature information is kept (or mostly kept), but in a space of smaller dimensionality. A good example of this is PCA, which transforms the space to a new space in which each dimension is a linear combination of the original features.
I do understand the confusion with the name though, if you have any suggestions for it let me know :)

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+        "selection": "study_based",
+        "role": "main",
+        "main_goal": "anomaly_detection",
+        "active_timeout": "missing",
```
Although they don't call it active timeout, they mention it is 5 minutes (300 seconds): "All histograms correspond to five
minute traces collected at the same time of the day."

fm94 replied:
But one histogram consist of multiple flows, so they are gathering flows of time-out 'x' for 300 seconds and calculate their histogram. It is kind of a histogram time-out and not a flows time-out.

dcferreira replied:
Ah, right. I think it's highly likely that the active time-out they use in their tool is larger than 300s, which would make the effective active time-out 300s. However, since they don't mention it, it's probably best to leave it as missing.

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+        "name": "hierarchical cluster analysis",
+        "subname": "hca",
+        "learning": "unsupervised",
+        "role": "intermediate",
```
Hierarchical clustering is the ``main`` method in this paper (not ``intermediate``). Actually, even though they mention k-means, I can't find where they use it.

fm94 replied:
That is true, I was confused and I thought they used k-means for clustering and hca for classification.

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+  },
+  "evaluation": {
+    "algorithm_comparison": false,
+    "internal_validation": true,
```
You say they use internal validation, but present no internal method. Could you add the method you mean? (or set this to false, if it was a mistake)

fm94 replied:
They used clustering to label the histograms but I am not sure if HCA is correct (IV.C)

dcferreira replied:
Not sure I understand what you mean. This section is for the metrics used to evaluate the methods, not for the methods themselves. Meaning, if you use for example K-means and only evaluate its accuracy, there is no internal validation. Internal validation would be metrics that don't use labels (e.g., silhouette).

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+    "external_validation": true,
+    "dpi-based_validation": false,
+    "port-based_validation": false,
+    "pre-knowledge-based_validation": false,
```
At least for the IDEval dataset they used ``pre-knowledge-based_validation``. I'm not sure how they got the labels for the other datasets though.

fm94 replied:
What do yo suggest? (for IDEval they had labels only for flows and not for histograms)

dcferreira replied:
Although I didn't explicitly read this, my guess is that they label histograms as anomalous if there is an anomalous flow in them, correct? If so, I'd say they use ``pre-knowledge-based_validation`` for the IDEval.

fm94 replied:
but for the other data-sets they didn't mention anything, I'll keep it like this for now

dcferreira commented on v2_papers/2009/kind_histogram.json:
```diff
@@ -0,0 +1,179 @@
+  },
+  "result": {
+    "main_goal": "anomaly_detection",
+    "focus_main": "_new_approach",
```
I'd say the main focus is the methodology, so the ``methodology/framework`` value.

fm94 replied:
The goal from the beginning was to detect anomalies, they presented a way to represent the data via histograms and then detect abnormal ones. What do you think ?

dcferreira replied:
I think "methodology" would be an appropriate way to describe what you said :)

dcferreira reviewed (commented):
Here are some comments regarding the histogram-based paper.
Note that these are mostly subjective, from someone who didn't read the paper very deeply, so if you don't agree with anything feel free to ignore it @fm94 :)

fm94 committed:
Update lim_internet.json

1 - author name fixed
2 - bibtex updated
3 - changed "tcpCwrTotalCount" to "_tcpCwrTotalCount",
4 - typo in "curated_by" fixed

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,287 @@
+      "booktitle": "Proceedings of the 6th International COnference",
+      "organization": "ACM"
+    },
+    "access_open": true,
```
Unfortunately this is not open access. If you go to https://dl.acm.org/citation.cfm?id=1921180 (while outside the university network) it will ask you to pay to access the paper.

fm94 replied:
from other search engines you can find the free pdf but I'll change it to false if we are talking about the ACM website

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,287 @@
+        "anonymized": true
+      },
+      {
+        "name": "WIDE-II",
```
Regarding WIDE-II (and I assume WIDE-I is the same), there are two versions:
1. one that is anonymized, without payloads, and public (http://imdc.datcat.org/collection/1-055M-0=WIDE-TRANSIT+100+Megabit+Ethernet+Trace+2007-01-09+%28Anonymized%29)
2. one that is not anonymized, has the first 96 bytes of each packet, and is available on request (http://imdc.datcat.org/collection/1-055N-M=WIDE-TRANSIT-100-Megabit-Ethernet-Trace-2007-01-09)

Since they don't cite the datasets it's hard to find out which version it is exactly, but I'd say it's the first one since they require the payloads. If you agree, change ``availability`` to ``public_on_demand`` and ``anonymized`` to ``false``.
I guess for WIDE-I it's exactly the same, but I'm only able to find the public version (anonymized, without payloads): http://mawi.wide.ad.jp/mawi/samplepoint-B/20060303/

fm94 replied:
I took the informations from the paper and I didn't check from the website, I'll do the changes

dcferreira commented on v2_papers/2010/lim_internet.json:
```diff
@@ -0,0 +1,287 @@
+      "new_phenomena_disclosed",
+      "improved_traffic_classification"
+    ]
+  }
```
Although it's not mandatory, I'd encourage you to add the ``reproducibility`` field :)

dcferreira reviewed (commented):
These are some suggestions regarding the ``lim_internet`` paper.

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+      "Comăneci, Dragoş",
+      "Dobre, Ciprian"
+    ],
+    "publication_name": "Management of the Internet of things and big data",
```
The actual journal name is "International Journal of Network Management".

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+    ],
+    "publication_name": "Management of the Internet of things and big data",
+    "publication_type": "peer_reviewed_journal",
+    "year": 2016,
```
The paper was published in 2017, from what I can find. (this also involves changing the file to the ``2017`` directory)

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+    "algorithms": [
+      {
+        "name": "Clustering",
+        "subname": "K-means",
```
Just for a matter of consistency within the database, can you instead have the ``name`` be "K-means"? (and I guess you can remove the subname)

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+            "add": [
+              {
+                "apply": [
+                  "octetTotalCount",
```
The ``add`` is not necessary here. By just having the apply, you'll get the ``octetTotalCount`` of the first 10 packets; that is, you'll get the sum of their size.

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+          "__first10senders",
+          {
+            "add": [
+              {
```
I'm not completely sure, but I think they're using 10 features for each of these "the first 10 ______". That is, I think each of these features corresponds to a vector in R^10.
If my interpretation is correct, here is how I'd write these features:
```
get(0, map(flowDirection, select(true)));
get(1, map(flowDirection, select(true)));
get(2, map(flowDirection, select(true)));
get(3, map(flowDirection, select(true)));
get(4, map(flowDirection, select(true)));
get(5, map(flowDirection, select(true)));
get(6, map(flowDirection, select(true)));
get(7, map(flowDirection, select(true)));
get(8, map(flowDirection, select(true)));
get(9, map(flowDirection, select(true)));
get(0, map(_interPacketTimeMicroseconds, select(true)));
get(1, map(_interPacketTimeMicroseconds, select(true)));
get(2, map(_interPacketTimeMicroseconds, select(true)));
get(3, map(_interPacketTimeMicroseconds, select(true)));
get(4, map(_interPacketTimeMicroseconds, select(true)));
get(5, map(_interPacketTimeMicroseconds, select(true)));
get(6, map(_interPacketTimeMicroseconds, select(true)));
get(7, map(_interPacketTimeMicroseconds, select(true)));
get(8, map(_interPacketTimeMicroseconds, select(true)));
get(9, map(_interPacketTimeMicroseconds, select(true)));
get(0, map(ipTotalLength, select(true)));
get(1, map(ipTotalLength, select(true)));
get(2, map(ipTotalLength, select(true)));
get(3, map(ipTotalLength, select(true)));
get(4, map(ipTotalLength, select(true)));
get(5, map(ipTotalLength, select(true)));
get(6, map(ipTotalLength, select(true)));
get(7, map(ipTotalLength, select(true)));
get(8, map(ipTotalLength, select(true)));
get(9, map(ipTotalLength, select(true)));
get(0, map(_interPacketTimeMicroseconds, select(true))) - get(1, map(_interPacketTimeMicroseconds, select(true)));
get(1, map(_interPacketTimeMicroseconds, select(true))) - get(2, map(_interPacketTimeMicroseconds, select(true)));
get(2, map(_interPacketTimeMicroseconds, select(true))) - get(3, map(_interPacketTimeMicroseconds, select(true)));
get(3, map(_interPacketTimeMicroseconds, select(true))) - get(4, map(_interPacketTimeMicroseconds, select(true)));
get(4, map(_interPacketTimeMicroseconds, select(true))) - get(5, map(_interPacketTimeMicroseconds, select(true)));
get(5, map(_interPacketTimeMicroseconds, select(true))) - get(6, map(_interPacketTimeMicroseconds, select(true)));
get(6, map(_interPacketTimeMicroseconds, select(true))) - get(7, map(_interPacketTimeMicroseconds, select(true)));
get(7, map(_interPacketTimeMicroseconds, select(true))) - get(8, map(_interPacketTimeMicroseconds, select(true)));
get(8, map(_interPacketTimeMicroseconds, select(true))) - get(9, map(_interPacketTimeMicroseconds, select(true)));
get(9, map(_interPacketTimeMicroseconds, select(true))) - get(10, map(_interPacketTimeMicroseconds, select(true)));
```

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+        "name": "K-means clustering",
+        "type": "internal",
+        "source": "own_proposed"
+      },
```
Did I miss something or is this a mistake? Is there an (internal) evaluation metric called "K-means clustering"?

fm94 replied:
They used K-means to label the data and then compare it with other results, this is what I meant

dcferreira replied:
See the new comment below. I don't see K-means being used as a metric, but I do see it being evaluated with accuracy.

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+  "evaluation": {
+    "algorithm_comparison": false,
+    "internal_validation": true,
+    "external_validation": false,
```
There definitely exists some external validation, and I didn't find any internal validation (see comments below).

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+        "name": "normal classification metrics",
+        "type": "internal",
+        "metrics": [
+          "accuracy"
```
Accuracy is definitely an ``external`` metric.

fm94 replied:
but labels were retrieved internally

dcferreira replied:
From what I read, the accuracy is always referring to the labels (in Table 1).
I really hope they aren't using the clusters as evaluation labels, because that would be completely meaningless (e.g., use just 1 cluster and you're guaranteed to have 100% accuracy). But I'm pretty sure they use the actual labels from their traffic generator.

fm94 replied:
That is what I thought also, I will change it

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+    "external_validation": false,
+    "dpi-based_validation": false,
+    "port-based_validation": false,
+    "pre-knowledge-based_validation": false,
```
I'd say it's ``pre-knowledge-based_validation``, since they have the labels from generating the data.

dcferreira commented on v2_papers/2016/vlăduţu_internet.json:
```diff
@@ -0,0 +1,306 @@
+      "application_classification"
+    ],
+    "focus_main": "methodology/framework",
+    "claimed_improvements": [],
```
From their conclusions, I'd add ``new_phenomena_disclosed`` to the conclusions.

dcferreira reviewed (commented):
Here are some comments on the last paper.

I was a bit heavy handed in this PR since it's your first time curating papers, the next times it should go a lot more smoothly.
When you've considered all the comments and think this is ready to merge (for all the papers), let me know and I'll do it :)

fm94 committed:
Update kind_histogram.json 

fixed :
1 - feature selections type
2 - clustering algorithm properties 
3 - internal validation set to false
4 - change focus main to methodology/framework

fm94 committed:
update lim_internet.json

fixed :
1 - access is not open
2 - data-sets are available on demand
3 - reproducibility field is added

fm94 commented:
Please read the comments and let me know to commit the changes

fm94 committed:
Update vlăduţu_internet.json

1 - fixed journal name and date
2 - edited the algorithm for the sake of consistency with the database
3 - fixed some non-IANA features
4 - fixed validation methods
5 - "new_phenomena_disclosed" is added to conclusion

fm94 commented:
vlăduţu_internet.json needs to be moved into 2017

dcferreira commented:
Can you move it into 2017? I can merge it right after.

fm94 committed:
Moved vlăduţu_internet.json from 2016 to 2017
